### PR TITLE
feat: Change price and solo listing modal

### LINKED ISSUE #7026: Change price and solo listing modal
image

- reuses a lot of components from the confirmation modal and 
- #6941 
- displaying collection floor 
- Change price only changes the heading in the top part and button copy 
- Floor price and % selectors have the same hover effect and styling as those in mass listing 
image

If you have any questions, feel free to ask! 

figma file: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=1458%3A19091&mode=design&t=mMjhohc2g1glvuP8-1

### LINKED ISSUE #7168: Allow user to change price and list at the same time when using masslisting
### What happened?

Goes after: #7076 

When trying two different options(list and change price) at the same time: 

https://github.com/kodadot/nft-gallery/assets/36627808/dce847f0-a734-4c9e-8b52-665ece16e1a4

### PR DESCRIPTION
**Pull request by hassnian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #7026
- [x] Closes #7168

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-09-10 at 17 24 42@2x

item gallery 

CleanShot 2023-09-12 at 15 31 41

### <samp>🤖 Generated by Copilot at cb8460a</samp>

This pull request adds new components and logic to handle the price input and display for the listing cart modal. It splits the `ListingCartModal` component into two subcomponents: `ListingCartSingleItemCart` and `ListingCartMultipleItemsCart`, depending on the number of items in the cart. It also refactors the `ListingCartPriceInput` component and creates new shared components: `ListingCartFloorPrice` and `ListingCartItemDetails`. Additionally, it deletes the old `ListingCartItem` component and adds new translation keys to the `en.json` file.

### <samp>🤖 Generated by Copilot at cb8460a</samp>

> _We're listing NFTs on the cart modal, me hearties_
> _We're using `ListingCartPriceInput` for the price_
> _We're showing the floor and the collection, me hearties_
> _We're heaving away on the count of three_

### CODE REVIEW

**Comment by hassnian** [comment]
@prury while testing this you might see some nft items not show or not appear inside the modal , just choose nfts with a floor price or wait for #7226 to be released and then merged

**Comment by exezbcz** [comment]
image

- not updating price inside of the input 
- the modal should popup also from the gallery item

image

- that means getting rid of this sliding animation

**Comment by exezbcz** [comment]
> @exezbcz 
> 
> > not updating price inside of the input
> 
> you mean the prev price is not set inside the input field right ? I did notice that but I thought it was by design , that's how it before , will fix it 
> 

When you click the floor buttons 


> > the modal should popup also from the gallery item
> 
> got it 
> 
>

**Comment by hassnian** [comment]
> > @exezbcz
> > > not updating price inside of the input
> > 
> > 
> > you mean the prev price is not set inside the input field right ? I did notice that but I thought it was by design , that's how it before , will fix it
> 
> When you click the floor buttons
> 
> > > the modal should popup also from the gallery item
> > 
> > 
> > got it

oooh got it i see

**Comment by hassnian** [comment]
added change price modal state 
CleanShot 2023-09-12 at 14 01 56@2x

**Comment by hassnian** [comment]
added item gallery 

CleanShot 2023-09-12 at 15 31 41

**Review by Jarsen136 — APPROVED** [review]

**Comment by prury** [comment]
List(design shows -- instead of 0): 
image

values inputted are not being reset: 

https://github.com/kodadot/nft-gallery/assets/36627808/2eaa7564-f2b3-417d-bbb0-3bce5a6f5d77

**Comment by hassnian** [comment]
@prury fixed

**Comment by prury** [comment]
> values inputted are not being reset:

this still happens ^

another bug(list modal keeps open even if you leave the page): 

https://github.com/kodadot/nft-gallery/assets/36627808/676afaf9-9825-4608-899c-1440589a01af

**Comment by exezbcz** [comment]
noticed this horizontal scroll, could you please fix it 

image

**Comment by hassnian** [comment]
@prury now it should be fixed

**Comment by prury** [comment]
> @prury now it should be fixed

cool, it works! 
Horizontal scroll pointed by Exez still there tho

**Review by roiLeo — APPROVED** [review]
✅ lgtm

File: components/common/listingCart/shared/ListingCartPriceInput.vue
Comment (hassnian) [reply components/common/listingCart/shared/ListingCartPriceInput.vue]: does the NeoInput component provide a way of adding a right section like this one 

CleanShot 2023-09-14 at 10 42 52@2x

renamed the class to 
CleanShot 2023-09-14 at 10 43 48@2x

w-full would need to be added in two places i'd rather have like this if it's ok

**Comment by hassnian** [comment]
@prury regarding the horizontal scroll could you make a video showing what element is making the modal have a scroll , is not showing up for me , don't want to clip the entire modal i'd rather fix that element

**Comment by prury** [comment]
> @prury regarding the horizontal scroll could you make a video showing what element is making the modal have a scroll , is not showing up for me , don't want to clip the entire modal i'd rather fix that element

its both change price and list, hehe, they are also being cut a bit by the navbar on the upper part
image

image

**Comment by roiLeo** [comment]
> its both change price and list, hehe, they are also being cut a bit by the navbar on the upper part

Looks like Windows issue, can't reproduce on OSX. something with a fixed width is making the modal overflow x axis

**Comment by prury** [comment]
> > its both change price and list, hehe, they are also being cut a bit by the navbar on the upper part
> 
> Looks like Windows issue, can't reproduce on OSX. something with a fixed width is making the modal overflow x axis

Proceed then?

**Review by prury — APPROVED** [review]

**Comment by roiLeo** [comment]
Capture d’écran 2023-09-21 à 4 26 53 PM

I'll merge it, ping yangwao if if it needs to be paid

**Comment by exezbcz** [comment]
yeah lets not hold it

**Comment by hassnian** [comment]
@yangwao payout missing

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.03 USD/DOT ~ 12.407 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0x284a3a525ea91621ee965531e9d22af431541fb4bbf108eb9a929e8b13856bd3](https://polkadot.subscan.io/extrinsic/0x284a3a525ea91621ee965531e9d22af431541fb4bbf108eb9a929e8b13856bd3)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**
